File: app/src/main/java/com/amarnehsoft/holyquran/main/readersFragment/ReadersAdapter.java
package com.amarnehsoft.holyquran.main.readersFragment;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.amarnehsoft.holyquran.R;
import com.amarnehsoft.holyquran.test.Reader;
import com.amarnehsoft.holyquran.utils.CircleTransform;
import com.squareup.picasso.Picasso;

import java.util.List;

public class ReadersAdapter extends RecyclerView.Adapter<ReaderHolder> {
    private List<Reader> readers;

    private LayoutInflater inflater;
    private Listener listener;

    ReadersAdapter(LayoutInflater inflater, Listener listener){
        this.inflater = inflater;
        this.listener = listener;
        this.readers = ReadersProvider.getReaders();
    }

    @NonNull
    @Override
    public ReaderHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = inflater.inflate(R.layout.row_reader, viewGroup, false);
        return new ReaderHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ReaderHolder holder, int i) {
        Reader reader = readers.get(i);
        holder.txtName.setText(reader.getName());
        if (!TextUtils.isEmpty(reader.getImgUrl())){
            Picasso.get().load(reader.getImgUrl()).transform(new CircleTransform()).into(holder.img);
        }

        holder.itemView.setOnClickListener(view -> {
            if (ReaderController.getCurrentReader() != reader){
                listener.onReaderClicked(reader);
            }
        });

    }

    @Override
    public int getItemCount() {
        return readers.size();
    }

    public interface Listener{
        void onReaderClicked(Reader reader);
    }
}

File: app/src/main/java/com/amarnehsoft/holyquran/main/readersFragment/ReadersAdapter.kt
package com.amarnehsoft.holyquran.main.readersFragment

import androidx.recyclerview.widget.RecyclerView
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.ViewGroup

import com.amarnehsoft.holyquran.R
import com.amarnehsoft.holyquran.test.Reader
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions

class ReadersAdapter internal constructor(
    private val inflater: LayoutInflater,
    private val listener: Listener
) : RecyclerView.Adapter<ReaderHolder>() {
    private val readers: List<Reader> = ReadersProvider.readers

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ReaderHolder {
        val view = inflater.inflate(R.layout.row_reader, viewGroup, false)
        return ReaderHolder(view)
    }

    override fun onBindViewHolder(holder: ReaderHolder, i: Int) {
        val reader = readers[i]
        holder.txtName.text = reader.shortName
        if (!TextUtils.isEmpty(reader.imgUrl)) {
            Glide.with(holder.itemView.context).load(reader.imgUrl).apply(RequestOptions.circleCropTransform())
                .into(holder.img)
        }

        holder.itemView.setOnClickListener { view ->
            if (ReaderController.currentReader !== reader) {
                listener.onReaderClicked(reader)
            }
        }

    }

    override fun getItemCount(): Int {
        return readers.size
    }

    interface Listener {
        fun onReaderClicked(reader: Reader)
    }
}
